tests/integration_tests/functional/test_metrics.py

```diff
@@ -38,11 +267,10 @@ def _validate_metrics(metrics):
         "vsock",
         "entropy",
     ]
-
     if platform.machine() == "aarch64":
-        exp_keys.append("rtc")
+        FirecrackerMetricsSchema["required"].append("rtc")
```

Comment:
do I get it right that this verifies that all the top level objects are present, but it doesn't verify that all fields are present? Are they considered optional?

Reply:
"required" makes sure that all fields of FirecrackerMetrics and all fields of its individual components are present and they are numbers. Right now, basically, we expect metrics to have everything that is there in FirecrackerMetricsSchema["properties"].
But I might have missed RTCDeviceMetrics since I tested this on x86 machine.

Reply:
fixed, thanks for catching this !

Reply:
converting to draft to try some more negative test cases.

Reply:
Done, added an extra test to make sure we capture breaking change of individual metrics fields.